### PR TITLE
few misc fixes, moved job milestone props to separate from applicant …

### PR DESCRIPTION
**Pull request by coryatwater-FW**
main change is to have the Recruitment component hold the state for jobs and milestones instead of applicant page holding it

### CODE REVIEW

File: apps/web/src/services/constants/milestone.constants.ts [hunk 1]
```diff
@@ -3,4 +3,8 @@ export const milestoneRecruitmentOptions = [
   { value: '25', label: 'Interview Completed' },
   { value: '26', label: 'References Completed' },
   { value: '27', label: 'Offered Position' },
+  { value: '28', label: 'Candidate Withdrew' },
+  { value: '29', label: 'Candidate Was Unresponsive' },
+  { value: '30', label: 'Candidate Was Not Selected' },
+  { value: '31', label: 'Candidate Accepted the Job Offer' },
```
Comment (bhumin-fw) [inline apps/web/src/services/constants/milestone.constants.ts:9]: Milestone values need to fetch from the database too. We can filter data for recruitment milestones and keep only required data in a database for it.

we have an API available for it. /api/v1/ienmaster/status
In this API you can fetch data for "id": 3, below the object.
{
"id": 3,
 "status": "IEN Recruitment (once licensed)"
 "children" : [ <array of an available milestone for the recruitment>]
}

If we push this data on Dev, in that database you will see different values for these milestones as these are auto-generated IDs.
if that is some big change here, we will plan in next PR. Let me know your thoughts.

**Review by bhumin-fw — APPROVED** [review]

File: apps/web/src/components/display/AddRecordModal.tsx [hunk 2]
```diff
@@ -42,11 +31,11 @@ export const AddRecordModal: React.FC<AddRecordProps> = ({ jobRecords, setJobRec
   useEffect(() => {
     try {
       const getRecordListData = async () => {
-        const data = await getAddRecordOptions();
+        const [haPcn, jobTitle, jobLocation] = await getAddRecordOptions();
```
Comment (dbayly-freshworks) [inline apps/web/src/components/display/AddRecordModal.tsx:34]: This is good. Much more readable!

File: apps/web/src/services/master-lists.ts [hunk 3]
```diff
@@ -9,3 +21,21 @@ export const getAddRecordOptions = async () => {
...
+
+export interface MilestoneType {
+  status: MilestoneTypeOptions[];
+}
+
+// milestone status' for adding milestones
+export const getMilestoneOptions = async () => {
+  const {
+    data: { data },
+  } = await axios.get('ienmaster/status');
```
Comment (dbayly-freshworks) [inline apps/web/src/services/master-lists.ts:38]: Can we add types to this?

**Review by dbayly-freshworks — APPROVED** [review]
A few comments but overall looks good.

**Pull request merged**